applications/MeshingApplication/custom_utilities/parmmg/pmmg_utilities.cpp

```diff
@@ -49,154 +50,347 @@
+
+                // Setting inside flag
+                if (Ref == 2) {
+                    p_element->Set(INSIDE, true);
+                } else if (Ref == 3) {
+                    p_element->Set(INSIDE, false);
+                    if (GetRemoveRegions()) p_element->Set(TO_ERASE, true);
+                }
+            }
+        }
+    } else {
+        // The existence of a _nullptr_ indicates a missing element. Two options are possible: error or replacement
+        Properties::Pointer p_prop = nullptr;
+        Element::Pointer p_base_element = nullptr;
+
+        // Sometimes PMMG creates elements where there are not, then we skip
+        if (rMapPointersRefElement[Ref].get() == nullptr) {
+            return p_element;
+        } else {
+            p_base_element = rMapPointersRefElement[Ref];
+            p_prop = p_base_element->pGetProperties();
+        }
+
+        // FIXME: This is not the correct solution to the problem, I asked in the PMMG Forum
+        if (mLocalToGlobalNodePostMap[vertex_0] == 0) SkipCreation = true;
+        if (mLocalToGlobalNodePostMap[vertex_1] == 0) SkipCreation = true;
+        if (mLocalToGlobalNodePostMap[vertex_2] == 0) SkipCreation = true;
+        if (mLocalToGlobalNodePostMap[vertex_3] == 0) SkipCreation = true;
+
+        if (!SkipCreation) {
+            std::vector<NodeType::Pointer> element_nodes (4);
+            element_nodes[0] = rModelPart.pGetNode(mLocalToGlobalNodePostMap[vertex_0]);
+            element_nodes[1] = rModelPart.pGetNode(mLocalToGlobalNodePostMap[vertex_1]);
+            element_nodes[2] = rModelPart.pGetNode(mLocalToGlobalNodePostMap[vertex_2]);
+            element_nodes[3] = rModelPart.pGetNode(mLocalToGlobalNodePostMap[vertex_3]);
+
+            p_element = p_base_element->Create(ElemId, PointerVector<NodeType>{element_nodes}, p_prop);
+        } else if (GetEchoLevel() > 2)
+            KRATOS_WARNING_IF("ParMmgUtilities", GetEchoLevel() > 1) << "Element creation avoided" << std::endl;
+    }
+
+    if (p_element!= nullptr) KRATOS_ERROR_IF(p_element->GetGeometry().Volume() < ZeroTolerance) << "Creating a almost zero or negative volume element" << std::endl;
+    return p_element;
 }
 
 /***********************************************************************************/
 /***********************************************************************************/
 
 template<>
-void ParMmgUtilities<PMMGLibrary::PMMG3D>::BlockCondition(const IndexType iCondition)
+void ParMmgUtilities<PMMGLibrary::PMMG3D>::InitMesh()
 {
-    KRATOS_ERROR_IF(PMMG_Set_requiredTriangle(mParMmgMesh, iCondition) != 1 ) << "Unable to block edge" << std::endl;
+    mParMmgMesh = nullptr;
+
+    // We init the PMMG mesh and sol
+    if (GetDiscretization() == DiscretizationOption::STANDARD) {
+        PMMG_Init_parMesh( PMMG_ARG_start, PMMG_ARG_ppParMesh, &mParMmgMesh, PMMG_ARG_pMesh, PMMG_ARG_pMet, PMMG_ARG_dim, 3, PMMG_ARG_MPIComm, MPI_COMM_WORLD, PMMG_ARG_end);
```

Comment:
@marcnunezc I came across this by chance and wanted to let you know that we should not use MPI_COMM_WORLD directly (any more since we have the MPI-Core)

Instead this should come from the DataCommunicator of the corresponding ModelPart by using [this](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/mpi/includes/mpi_data_communicator.h#L248) method

Could you please change this? Not urgent
Thx!

Reply:
Ok! Added to the ParMMg after release to-do in #7728